src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs

```diff
@@ -222,6 +236,39 @@ private static GitPullRequest CreatePullRequest(string sourceBranch, string targ
             throw new InvalidOperationException("Could not find drop path");
         }
 
+        private static async Task<string> DownloadArtifactsAsync(BuildHttpClient buildClient, Build build, BuildArtifact artifact, CancellationToken cancellationToken)
+        {
+            var tempDirectory = Path.Combine(Path.GetTempPath(), string.Concat(Options.InsertionName, Options.BranchName).Replace(" ", "_").Replace("/","_"));
+            if (Directory.Exists(tempDirectory))
```

Comment:
I still think this is suspicious.

Doing:
```
if (Directory.Exists(tempDirectory))
{
    Directory.Delete(path, recursive: true);
}
Directory.CreateDirectory(path);
```
is pretty standard practice.

and if there really is a race condition, we could add some basic retry logic or also cleanup after the insertion completes (rather than just at the start of the current insertion).
* If there is a race condition, that also means that these folders could (theoretically) not be deleted by the time we start this insertion (and that could put us in a bad state)

Reply:
hmm, maybe [this other github issue](https://github.com/dotnet/corefx/issues/15330) might help. Looks like we would need to upgrade to a later .Net framework to get this working(4.7.1 maybe). I could try that out, wonder if the scheduler machine has that installed though?

Reply:
@jasonmalinowski, what are your thoughts here?

Reply:
I'm OK upgrading if we think there's an actual fix (but I didn't see an actual fix with that issue?)

In any case, it seems the simpler code would just be:

```c#
while (Directory.Exists(tempDirectory))
try
{
    Directory.Delete(tempDirectory, recursive: true)
}
catch (Exception e) {}
```

or the likes. No reason to invent your first round of recursion here.

Reply:
Fixed.